app/javascript/angular/code/services/google/_map.js

```diff
@@ -10,17 +10,19 @@ export const map = (
   rectangles,
   geocoder,
   googleMaps,
+  $window,
   heat
 ) => {
   const TIMEOUT_DELAY = process.env.NODE_ENV === "test" ? 0 : 1000;
   let hasChangedProgrammatically = false;
+  $window.__markers = [];
```

Comment:
Using window as a tool to decouple stuff. At some point we will be able to remove things from there